staging/src/k8s.io/client-go/util/certificate/certificate_manager.go

```diff
@@ -261,7 +266,8 @@ func (m *manager) Start() {
 			case <-timer.C:
 				// unblock when deadline expires
 			case <-templateChanged:
-				if reflect.DeepEqual(m.getLastRequest(), m.getTemplate()) {
+				_, lastRequestTemplate := m.getLastRequest()
+				if reflect.DeepEqual(lastRequestTemplate, m.getTemplate()) {
```

Comment:
It would be nice if we didn't reason about certs here and just did

```
go wait.Until(func() {
  ctx, cancel : context.WithCancel(ctx)
  defer cancel()
  go wait.ExponentionBackoff(m.rotateCerts(ctx))

  select {
  case <-templateChange:
  case <-timerChange:
  }
})
```

and move the checks like:
```
if m.getTemplate() == nil {
```
and
```
if reflect.DeepEqual(m.getLastRequest(), m.getTemplate()) {
```

Into rotateCerts. Then all this loop does is trigger restarts of rotateCerts. rotateCerts handles all the details of the cert (if it shouldn't do anything yet because template is nil, if the previously created CSR is sufficient to allow a resumption a WaitForCertificate, etc...).

Reply:
I don't disagree this could be better structured, but I'd like to keep this PR targeted on fixing the 15-minute delay, and I don't think it makes it harder to do the kind of cleanup you describe as a follow-up